sigstore/_cli.py

```diff
@@ -468,6 +469,33 @@ def _verify(args: argparse.Namespace) -> None:
                     ),
                     file=sys.stderr,
                 )
+            elif isinstance(result, RekorEntryMissing):
+                # If Rekor lookup failed, it's because the certificate either
+                # wasn't logged after creation or because the user requested the
+                # wrong Rekor instance (e.g., staging instead of production).
+                # The latter is significantly more likely, so we add
+                # some additional context to the output indicating it.
+                #
+                # NOTE: Even though the latter is more likely, it's still extremely
+                # unlikely that we'd hit this -- we should always fail with
+                # `CertificateVerificationFailure` instead, as the cert store should
+                # fail to validate due to a mismatch between the leaf and the trusted
+                # root + intermediates.
```

Comment:
Flagging for review: the likelihood of different failure modes is speculation on my part, here. I figure it's much less likely for a sigstore client to forget to publish to the tlog than for a user to accidentally hit the wrong tlog.

However, as the `NOTE:` says, it would still be pretty difficult for a user to hit this *particular* error case, as there would have to be an earlier bug in cert chain validation. To actually trigger this for testing, I had to disable our cert chain validation.

Reply:
I see: I just checked validation code, and I agree: in order for a user to hit this bug, it would somehow mean that the certificate pool was configured with the CORRECT root, but that the tlog was pointing to a different instance of the ecosystem.

A mismatch between the fulcio roots and tlog are unlikely I agree: both the incorrect tlog request & the certificate verification errors are pointing to errors in the client itself: the only time users would see this and it WOULDN'T be the client's fault is LIKELY a malicious signature event. 

IMO: It is worth a callout that this may actually be a failure like "The signing event could not be verified using the configured log.  This may be a result of connecting to the wrong Rekor instance..." like you say.

Reply:
Sounds good; I'll tweak to cover that!

Reply:
lgtm!